test_requests_resource.py

```diff
@@ -0,0 +1,197 @@
+        empty_string = self.client.post(
+            'api/v1/users/requests/', data="", content_type="application/json")
+        bad_data = self.client.post('api/v1/users/requests/', data={
+            "request_id": 1,
+            'title': '',
+            "location": 234,
+        }, content_type="application/json"
+        )
+
+        self.assertEqual(empty_dict.status_code, 400)
+        self.assertEqual(empty_tuple.status_code, 400)
+        self.assertEqual(empty_list.status_code, 400)
+        self.assertEqual(empty_string.status_code, 400)
+        self.assertEqual(bad_data.status_code, 400)
+
+    def test_update_a_request(self):
+        """ Test a resource can be successfully updated """
+        self.register_and_authenticate_user()
+
+        response = self.client.post(
+            'api/v1/users/requests/', data=self.data["request"],
+            content_type="application/json"
+        )
+        self.assertEqual(response.status_code, 201)
+        created_resource = json.loads(response.data)
+        response = self.client.put(
+            f'api/v1/users/requests/{created_resource["request_id"]}/', data={
+                "title": "My Actual request",
+                "request_type": "repair"
+            },
+            content_type="application/json"
+        )
+        self.assertEqual(response.status_code, 200)
+
+    def test_400_update_a_request(self):
+        """ Test bad request on put method """
+        self.register_and_authenticate_user()
+
+        response = self.client.post(
+            'api/v1/users/requests/', data=self.data["request"],
+            content_type="application/json"
+        )
+        self.assertEqual(response.status_code, 201)
+
+        created_resource = json.loads(response.data)
+
+        empty_dict = self.client.put(
+            f'api/v1/users/requests/{created_resource["request_id"]}/', data={},
+            content_type="application/json")
+        empty_tuple = self.client.put(
+            f'api/v1/users/requests/{created_resource["request_id"]}/', data=(),
+            content_type="application/json")
+        empty_list = self.client.put(
+            f'api/v1/users/requests/{created_resource["request_id"]}/', data=[],
+            content_type="application/json")
+        empty_string = self.client.put(
+            f'api/v1/users/requests/{created_resource["request_id"]}/', data="",
+            content_type="application/json")
+        bad_data = self.client.put(f'api/v1/users/requests/{created_resource["request_id"]}/', data={
+            "request_id": 1,
+            'title': '',
+            "location": 234,
+        }, content_type="application/json")
+
+        self.assertEqual(empty_dict.status_code, 400)
+        self.assertEqual(empty_tuple.status_code, 400)
+        self.assertEqual(empty_list.status_code, 400)
+        self.assertEqual(empty_string.status_code, 400)
+        self.assertEqual(bad_data.status_code, 400)
+
+    def test_delete_a_request(self):
```

Comment:
This test can be split into several tests: eg `test_delete_a_request` , `tests_delete_a_request_that_does_not_exist` etc

Reply:
Thank you it makes a lot of sense because it gets cleaner and is readable and organized.  Working on it ASAP.